File: src/main/java/com/example/common/entity/Comment.java
package com.example.common.entity;

import javax.persistence.*;
import java.util.List;

/**
 * Created by Anatol on 10.06.16.
 */
@Entity
@Table(name = "comment")
public class Comment extends BaseEntity {

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToOne
    @JoinColumn(name = "scheme_id", nullable = false)
    private Scheme scheme;

    @Column(nullable = false)
    private String commentary;

    @OneToMany(mappedBy = "commentId", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private List<Like> likes;

    public User getUser() {
        return user;
    }

    public List<Like> getLikes() {
        return likes;
    }

    public void setLikes(List<Like> likes) {
        this.likes = likes;
    }

    public User setUser(User user) {
        this.user = user;
        return user;
    }

    public Scheme getScheme() {
        return scheme;
    }

    public Scheme setScheme(Scheme scheme) {
        this.scheme = scheme;
        return scheme;
    }


    public String getCommentary() {
        return commentary;
    }

    public String setCommentary(String commentary) {
        this.commentary = commentary;
        return commentary;
    }
}

File: frontend/src/app/dto/Comment.ts
import User from "./User";
import Scheme from "./Scheme";
import {Like} from "./Like";
export class Comment {
  private id: number;
  private user: User;
  private scheme: Scheme;
  private commentary: string;
  private likes: Array<Like>;

  constructor(id:number, user:User, scheme:Scheme, commentary:string, likes:Array<Like>) {
      this.id = id;
      this.user = user;
      this.scheme = scheme;
      this.commentary = commentary;
      this.likes = likes;
  }
  
  getId():number{
    return this.id;
  }

  setId(value:number){
    this.id=value;
  }

  getUser():User{
      return this.user;
      }

  setUser(value:User){
      this.user=value;
      }

  getScheme():Scheme{
      return this.scheme;
      }

  setScheme(value:Scheme){
      this.scheme=value;
      }

  getCommentary():string{
      return this.commentary;
      }

  setCommentary(value:string){
      this.commentary=value;
      }

  get likes():Array<Like>{
      return this.likes;
      }

  set likes(value:Array<Like>){
      this.likes=value;
      }
}
